### PR TITLE
mcs: sync store state to scheduling service

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

This PR synchronizes the changed state to the scheduling service because the heartbeat only reports the statistics.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/meta/watcher.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+	return w, nil
+}
+
+func (w *Watcher) initializeStoreWatcher() error {
+	putFn := func(kv *mvccpb.KeyValue) error {
+		store := &metapb.Store{}
+		if err := proto.Unmarshal(kv.Value, store); err != nil {
+			log.Warn("failed to unmarshal store entry",
+				zap.String("event-kv-key", string(kv.Key)), zap.Error(err))
```
Comment (JmPotato) [inline pkg/mcs/scheduling/server/meta/watcher.go:76]: Do we need to call `ExtractStoreIDFromPath` to get the store ID here?

**Review by JmPotato — APPROVED** [review]
The reset LGTM.

**Review by lhy1024 — APPROVED** [review]

File: pkg/mcs/scheduling/server/meta/watcher.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/scheduling/server/meta/watcher.go:76]: Do you mean to add it to the log?

**Pull request merged**